hardware-api.md

```diff
@@ -91,7 +97,23 @@ pin.analogRead(function(error, value) {
 
 ### PWM pins
 
-PWM pins are not yet implemented. See [#21](https://github.com/tessel/t2-firmware/issues/21).
+**PWM pins** are pulse-width modulated pins ([wiki link](http://en.wikipedia.org/wiki/Pulse-width_modulation)). Essentially, PWM is a digital signal that spends between 0% and 100% of its time on (this is its "duty cycle"). You can set the PWM pins to any value between 0 and 1 to approximate an analog signal. PWM is often used to control servo speeds or LED brightness.
```

Comment:
not clear how this can be used to approximate an analog signal. Maybe give an example like "e.g. if you wanted an analog signal of 2V, that's about 60% of 3.3V, so you could set PWM to 0.6. This would mean it's all the way on 60% of the time (and all the way off 40% of the time), instead of the analog 60% on all of the time." Ugh that's a bit long but still might not answer the basic question of "when is it ok to do that?"